x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/EsqlSession.java

```diff
@@ -544,4 +630,52 @@ public PhysicalPlan optimizedPhysicalPlan(LogicalPlan optimizedPlan) {
         LOGGER.debug("Optimized physical plan:\n{}", plan);
         return plan;
     }
+
+    class IndexResolutionWrappingListener implements ActionListener<EnrichResolution> {
+
+        private final ActionListener<Tuple<EnrichResolution, IndexResolution>> delegate;
+        private final IndexResolution indexResolution;
+
+        IndexResolutionWrappingListener(
+            ActionListener<Tuple<EnrichResolution, IndexResolution>> delegate,
```

Comment:
Using a record instead of Tuple<En...> makes sense here and allows for future evolution without noise all over the file:
`record ListenerResult(EnrichResolution, IndexResolution,...)` since we're likely have more `Resolution` in the future